framework/src/mesh/StitchedMesh.C

```diff
@@ -0,0 +1,108 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "StitchedMesh.h"
+#include "Parser.h"
+#include "InputParameters.h"
+
+// libMesh includes
+#include "libmesh/mesh_modification.h"
+#include "libmesh/serial_mesh.h"
+#include "libmesh/exodusII_io.h"
+
+template<>
+InputParameters validParams<StitchedMesh>()
+{
+  InputParameters params = validParams<MooseMesh>();
+  params.addRequiredParam<std::vector<MeshFileName>>("files", "The name of the mesh files to read.  These mesh files will be 'stitched' into the current mesh in this order.");
+
+  params.addRequiredParam<std::vector<BoundaryName>>("stitch_boundaries", "Pairs of boundary names (one after the other) to stitch together for each step.");
+
+  params.addParam<bool>("clear_stitched_boundary_ids", true, "Whether or not to erase the boundary IDs after they've been used for stitching.");
+
+  params.addClassDescription("Reads in all of the given meshes and stitches them all together into one mesh.");
+
+  return params;
+}
+
+StitchedMesh::StitchedMesh(const InputParameters & parameters) :
+    MooseMesh(parameters),
+    _files(getParam<std::vector<MeshFileName> >("files")),
+    _clear_stitched_boundary_ids(getParam<bool>("clear_stitched_boundary_ids")),
+    _stitch_boundaries(getParam<std::vector<BoundaryName>>("stitch_boundaries"))
+{
+  // The StitchedMesh class only works with ReplicatedMesh
+  errorIfDistributedMesh("StitchedMesh");
+
+  // Get the original mesh
+  _original_mesh = dynamic_cast<ReplicatedMesh *>(&getMesh());
+  if (!_original_mesh)
+    mooseError("StitchedMesh does not support DistributedMesh");
+
+  // Read the first mesh into the original mesh... then we'll stitch all of the others into that
+  _original_mesh->read(_files[0]);
```

Comment:
Oh really?  Bummer.

I figured that since it was a `requiredParam` it would have at least one entry in it.

Is that a thing we can do in MOOSE-land (check that required vector params have at least one entry in them)?  It's really kind of wrong that it's allowed.

I'll add a test for `_files.size()`...